### PR TITLE
tz/298/uploads-by-expenditure-category

### PR DESCRIPTION
**Pull request by tzinckgraf**
Issue 298

Added function for uploads by expenditure category. Added test caes as well.

### CODE REVIEW

File: api/src/lib/aws.test.ts
Comment (tzinckgraf) [inline api/src/lib/aws.test.ts:1]: Not sure why this whole file is showing up as new.

File: api/src/lib/aws.ts [hunk 1]
```diff
@@ -97,13 +97,21 @@ async function sendHeadObjectToS3Bucket(bucketName: string, key: string) {
   await s3.send(new HeadObjectCommand(uploadParams))
 }
 
+export function getS3UploadFileKey(organizationId: number, upload: Upload | CreateUploadInput, uploadId?: number) {
+  if ('id' in upload) {
```
Comment (laurelkorwin) [inline api/src/lib/aws.ts:101]: Could we add a small explanatory comment here for when we'd expect this to hit vs the `uploadId` we pass in?

File: api/src/services/uploads/uploads.test.ts [hunk 2]
```diff
@@ -203,3 +211,45 @@ describe('downloads', () => {
     )
   })
 })
+
+describe('getUploads', () => {
+  scenario('returns most recent upload for one category',
+    async (scenario: UploadScenario) => {
+      mockCurrentUser(scenario.user.one)
+      await updateOrganization({
```
Comment (laurelkorwin) [inline api/src/services/uploads/uploads.test.ts:219]: Could we pull out these `await` calls here and in the scenario below into a `beforeEach` block?

File: api/src/services/uploads/uploads.ts [hunk 3]
```diff
@@ -147,3 +147,114 @@ export const Upload: UploadRelationResolvers = {
...
+      agency: { organizationId: organization.id },
+      id: { in: validUploadIds.map((v) => v.uploadId) },
+    },
+    include: { expenditureCategory: true },
+  })
+
+  const uploadsByExpenditureCategory = {}
+
+  // Get the most recent upload for each expenditure category and agency and set the S3 Object key
+  for (const upload of uploadsForPeriod)  {
```
Comment (laurelkorwin) [inline api/src/services/uploads/uploads.ts:225]: Ultimately up to you and totally fine in a follow-up, but wondering if a small wrapper class for `uploadsByExpenditureCategory` with some getters and setters might make it easier for future devs to understand / edit this code?

**Review by laurelkorwin — APPROVED** [review]

File: api/src/services/uploads/uploads.ts (hunk 3 above)
Comment (tzinckgraf) [reply api/src/services/uploads/uploads.ts:225]: Good idea, but let's do this in a follow up PR. I think this PR is a blocker on some other work.

**Review by laurelkorwin — APPROVED** [review]

**Pull request merged**